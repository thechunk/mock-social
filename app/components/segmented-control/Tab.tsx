import * as React from 'react';
import {PureComponent} from "react";
import {Text, TouchableOpacity, View} from 'react-native';
import {SegmentedControlContext} from "./SegmentedControl";

export interface ITab {
    id: number,
    title: string
}
export default class Tab extends PureComponent<ITab> {
    static contextType = SegmentedControlContext;

    constructor(props: ITab) {
        super(props);
        this.onPress = this.onPress.bind(this);
    }

    onPress() {
        this.context.setSelectedTab(this.props.id)
    }

    render() {
        return (
            <View>
                <TouchableOpacity onPress={this.onPress}>
                    <Text>{this.context.selectedTab === this.props.id ? 'active: ' : ''}{this.props.title}</Text>
                </TouchableOpacity>
            </View>
        );
    }
}

