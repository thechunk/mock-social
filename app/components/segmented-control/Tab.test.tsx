import * as React from 'react';
import {shallow} from "enzyme";
import Tab from "./Tab";

describe('Tab', () => {
    it('renders correctly', () => {
        expect(shallow(<Tab id={0} title="Title" />)).toMatchSnapshot();
    });
});