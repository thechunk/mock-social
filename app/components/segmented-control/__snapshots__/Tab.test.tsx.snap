// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      undefined,
      null,
    ]
  }
>
  <Text
    style={
      Array [
        undefined,
        null,
      ]
    }
  >
    Title
  </Text>
</TouchableOpacity>
`;
